src/devices/Bmxx80/Bmp280.cs

```diff
@@ -27,6 +27,8 @@ public Bmp280(I2cDevice i2cDevice)
             _communicationProtocol = CommunicationProtocol.I2c;
             _calibrationData = new Bmp280CalibrationData();
             _calibrationData.ReadFromDevice(this);
+
+            SetDefaultConfiguration();
```

Comment:
NIT: can we just add this call to the Bmxx80base constructor instead since presumably we want to set this for every device?

Reply:
Yes and no, as it currently stands calling it from the base would result in an Exception because the calibrationData is not yet read (which happens in the derived constructor) and `SetDefaultConfiguration` in the Bme680 needs some of those values. Plus I thought it would be bad practice in general to call a virtual method from the base constructor but I'm still fairly new to C# so I could be wrong.

Reading the calibration data in the base constructor would theoretically be possible and would allow us to call SetDefaultConfiguration after but would require some changes as well.

Reply:
Calling a virtual method from the base constructor should be fine in this case, and if you know that you need to perform some calibration data, you could also just have a ReadCalibrationData abstract method which you call from the base constructor, and define that on the subclasses with what calibration needs o be loaded. In general, the idea would be to bake in the most logic we can on the base class, and only have device specific code in the concrete devices/constructors, and seems like setting the default configuration is something that all devices would need to do.

Reply:
Alright changed it so that calibration data and default configuration gets loaded in the base.